presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -601,6 +611,25 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        List<Partition> partitions;
+        if (numSegments <= 1) {
+            partitions = getPartitions(databaseName, tableName, expression, null);
```

Comment:
```suggestion
            return getPartitions(databaseName, tableName, expression, null);
```

then, remove `else {` below, it will become redundant